WWDC/ImageDownloadCenter.swift

```diff
@@ -34,6 +34,7 @@ final class ImageDownloadCenter: Logging {
     
     func cachedThumbnail(from url: URL) -> NSImage? { cache.cachedImage(for: url, thumbnailOnly: true).thumbnail }
 
+    /// The completion handler is always called on the main thread
```

Comment:
Not directly related to this PR, but we should probably start annotating callbacks with `@MainActor` where applicable 😄